src/activities/test-prompts-activities.ts

```diff
@@ -0,0 +1,82 @@
+import fs from 'fs'
+
+import { createMemoizedOpenAI } from '../chat-gpt'
+
+const getGPTModel = createMemoizedOpenAI("gpt-4-turbo");
+
+type LoadTestCasesInput = {
+  testName: string
+}
+type LoadTestCasesOutput = Record<string, string>
+export async function loadTestCases(input: LoadTestCasesInput): Promise<LoadTestCasesOutput> {
+  const testCases = await fs.promises.readFile(`./test/prompts/${input.testName}.json`, 'utf8')
+  return JSON.parse(testCases)
+}
+
+type ValidateQueryResultInput = {
+  query: string
+  expectedResponse: string
+  actualResponse: string
+}
+type ValidateQueryResultOutput = {
+  query: string
+  answer: string
+  score: number
+  reason: string
+}
+export async function validateQueryResult(input: ValidateQueryResultInput): Promise<ValidateQueryResultOutput> {
+  const gptModel = getGPTModel()
+
+  const response = await gptModel.invoke([
+    [ 'system', 'You are responsible for verifying that prompt provided invoked the correct response from an LLM. Consider the following question and expected answer as well as the prompt\'s answered.' ],
```

Comment:
should `prompt's answered` be `prompt's answer`? It likely doesn't make a difference, just curious.